### PR TITLE
Makes fire colorful and anime

### PR DESCRIPTION
**Pull request by DaedalusGame**
:cl:
tweak: made fire colored according to blackbody radiation and rule of cool.
imageadd: some kind of lightning texture
/:cl:

https://i.imgur.com/ETmXnWk.gifv

Looks something like this.
- Makes fire additive blended, which means you can sometimes see through to the lower turf on colder fires
- Makes fire greyscaled using a color matrix and colors it the same way
- Colored fire lets you roughly gauge the temperature in mixing chambers.
- When fire gets "absurdly" hot it has special effect overlays added. (This may not be done very well, please review this)
- Renders fire on top of gas and pretty much everything else, alpha is adjusted for lower temperature fires so you can still see items through the fire.

Things that may still need doing:
- Ensure cutting and adding overlays isn't terrible
- ~~Modify light color without putting a strain on the server (Help appreciated)~~
- Gauss Lerp could be a define
- Code could be cleaned up probably
- ~~Updating the color and effects could be moved to a proc~~

Feedback welcome
Requires #34952 to be merged.

### CODE REVIEW

**Comment by ayyourboibilly** [comment]
anime

**Comment by DaedalusGame** [comment]
should i have put sonic flash instead

**Comment by Cyberboss** [comment]
Can you potentially update this less frequently? I'd hate to add this much code to every hotspot process

**Comment by Cyberboss** [comment]
Looks good though

**Comment by Tlaltecuhtli** [comment]
that picture isnt animated tho

**Comment by MoreRobustThanYou** [comment]
recompile this, lerp should be fixed

**Comment by WJohn** [comment]
This is really cool. Only thing to note is that hotter fires don't seem to obscure the floor any more than colder fires.

I'd like to see this "absurdly hot fire effect", too.

**Comment by MoreRobustThanYou** [comment]
@DaedalusGame 
```git remote add tg https://github.com/tgstation/tgstation.git
git pull tg master```

**Comment by Iamgoofball** [comment]
this does support reaching blue/white heat levels right

**Comment by DaedalusGame** [comment]
@WJohn 

400k Kelvin


5mil Kelvin (This one feels like it needs to be more violent.

@Iamgoofball I think white is somewhere pretty early like 6000 Kelvin and blue is past that.

**Comment by banthebantz** [comment]
You may be interested in the "heat2color" proc found here: https://github.com/ParadiseSS13/Paradise/blob/3eaff16f45583bd60366e96fdbc6bd63f29dc9c4/code/__HELPERS/type2type.dm

Seems a bit better than a bunch of ifs to me.

**Comment by RemieRichards** [comment]
@banthebantz he's using them already.

**Comment by DaedalusGame** [comment]
The ifs are just for a) putting the old fire texture to use at low temperature and b) making sure that fire above unreasonable temperatures isn't just flat blue (the algorithm is only defined up to 40000 Kelvin)

**Comment by WJohn** [comment]
swap out the 5 mil and 400 k and i think we're good.

**Comment by DaedalusGame** [comment]
Better idea, I'll change up the math a bit on going purple and make 5mil into rainbow

**Comment by DaedalusGame** [comment]
It's hard man, fire doesn't actually get that hot ingame without fusion and fusion is disabled

**Comment by DaedalusGame** [comment]
https://i.imgur.com/FE8NtDC.gifv

**Comment by DaedalusGame** [comment]
As stated in the commit, at request of Joan in coderbus, I've moved fire to a higher layer, to render above gases. Feedback on this change please.

https://i.imgur.com/ETmXnWk.gifv

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 1]
```diff
@@ -50,7 +50,7 @@
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	icon = 'icons/effects/fire.dmi'
 	icon_state = "1"
-	layer = ABOVE_OPEN_TURF_LAYER
+	layer = FLY_LAYER + 0.1//ABOVE_OPEN_TURF_LAYER
```
Comment (ShizCalev) [inline code/modules/atmospherics/environmental/LINDA_fire.dm]: RIPPLE_LAYER

**Review by ShizCalev — DISMISSED** [review]

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 1 above)
Comment (DaedalusGame) [reply code/modules/atmospherics/environmental/LINDA_fire.dm]: what the fuck is a hyperspace ripple
nevermind give me a second

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 2]
```diff
@@ -50,7 +50,7 @@
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	icon = 'icons/effects/fire.dmi'
 	icon_state = "1"
-	layer = FLY_LAYER + 0.1//ABOVE_OPEN_TURF_LAYER
+	layer = FIRE_LAYER //ABOVE_OPEN_TURF_LAYER
```
Comment (ShizCalev) [inline code/modules/atmospherics/environmental/LINDA_fire.dm]: can remove the comment too, if you wouldn't mind.

**Review by vuonojenmustaturska — DISMISSED** [review]

**Comment by vuonojenmustaturska** [comment]
testmerge seems to work, let's do it

**Pull request merged**